### PR TITLE
Implementations of listOpenWorkflow, listClosedWorkflow and terminateWorkflow

### PR DESCRIPTION
**Pull request by abhishekj720**

### CODE REVIEW

File: src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java [hunk 1]
```diff
@@ -190,6 +190,105 @@ public ListWorkflowExecutionsResponse ListWorkflowExecutions(
...
+                listRequest.getNextPageToken(),
+                _marker.length,
+                listRequest.getNextPageToken().length));
+      }
+      response = serviceNew.ListOpenWorkflowExecutions(listRequest);
+
+      if (response.getExecutionsSize() < listRequest.getMaximumPageSize()) {
+        int neededPageSize = listRequest.getMaximumPageSize() - response.getExecutionsSize();
+        ListOpenWorkflowExecutionsRequest copiedRequest = new ListOpenWorkflowExecutionsRequest();
+        copiedRequest = listRequest;
```
Comment (bowenxia) [inline src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java:219]: What is the point of doing a shallow copy here? The maximumPageSize of listRequest will also be changed.

File: src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java [hunk 2]
```diff
@@ -190,6 +190,105 @@ public ListWorkflowExecutionsResponse ListWorkflowExecutions(
...
+                _marker.length,
+                listRequest.getNextPageToken().length));
+      }
+      response = serviceNew.ListClosedWorkflowExecutions(listRequest);
+
+      if (response.getExecutionsSize() < listRequest.getMaximumPageSize()) {
+        int neededPageSize = listRequest.getMaximumPageSize() - response.getExecutionsSize();
+        ListClosedWorkflowExecutionsRequest copiedRequest =
+            new ListClosedWorkflowExecutionsRequest();
+        copiedRequest = listRequest;
```
Comment (bowenxia) [inline src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java:269]: same here

File: src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java [hunk 3]
```diff
@@ -190,6 +190,105 @@ public ListWorkflowExecutionsResponse ListWorkflowExecutions(
...
+
+      if (response.getExecutionsSize() < listRequest.getMaximumPageSize()) {
+        int neededPageSize = listRequest.getMaximumPageSize() - response.getExecutionsSize();
+        ListOpenWorkflowExecutionsRequest copiedRequest = new ListOpenWorkflowExecutionsRequest();
+        copiedRequest = listRequest;
+        copiedRequest.maximumPageSize = neededPageSize;
+        ListOpenWorkflowExecutionsResponse fromResponse =
+            serviceOld.ListOpenWorkflowExecutions(copiedRequest);
+
+        fromResponse.getExecutions().addAll(response.getExecutions());
```
[the remark this answers is not in the document]
Comment (abhishekj720) [reply src/main/java/com/uber/cadence/migration/MigrationIWorkflowService.java:224]: For client, it only contains the valid records in execution and hence, not require to do explicit check.

**Review by bowenxia — APPROVED** [review]
LGTM

**Pull request merged**